packages/neo-one-client-core/src/user/DapiUserAccountProvider.ts

```diff
@@ -0,0 +1,394 @@
+    const transaction = await this.provider.getTransaction(from.network, txid);
+    if (transaction.type !== 'ContractTransaction') {
+      throw new Error('For TS');
+    }
+
+    return {
+      transaction,
+      confirmed: async () => this.provider.getTransactionReceipt(from.network, txid),
+    };
+  }
+
+  protected async executeClaim(
+    _from: UserAccountID,
+    _attributes: readonly Attribute[],
+    _networkFee: BigNumber,
+  ): Promise<TransactionResult<TransactionReceipt, ClaimTransaction>> {
+    throw new NotImplementedError('executeClaim');
+  }
+
+  protected async executeInvokeScript<T extends TransactionReceipt>(
+    _options: ExecuteInvokeScriptOptions<T>,
+  ): Promise<TransactionResult<T, InvocationTransaction>> {
+    throw new NotImplementedError('executeInvokeScript');
+  }
+
+  protected async executeInvokeClaim(
+    _options: ExecuteInvokeClaimOptions,
+  ): Promise<TransactionResult<TransactionReceipt, ClaimTransaction>> {
+    throw new NotImplementedError('executeInvokeClaim');
+  }
+
+  protected async executeInvokeMethod<T extends TransactionReceipt>({
+    invokeMethodOptions,
+    from,
+    attributes,
+    inputs,
+    outputs,
+    rawInputs,
+    rawOutputs,
+    verify,
+    reorderOutputs,
+    onConfirm,
+  }: ExecuteInvokeMethodOptions<T>): Promise<TransactionResult<T, InvocationTransaction>> {
+    await this.initPromise;
+    const { contract, invokeMethod, params } = invokeMethodOptions;
+
+    const { txid } = await this.dapi.invoke({
+      scriptHash: addressToScriptHash(contract),
+      operation: invokeMethod,
+      args: params.map(paramToArgDataType),
+      network: this.defaultNetworkInternal$.getValue(),
+      triggerContractVerification: verify,
+      txHashAttributes: this.convertDapiAttributes(attributes),
+      assetIntentOverrides: {
+        inputs: this.convertDapiInputs(rawInputs.concat(inputs)),
+        outputs: this.convertDapiOutputs(reorderOutputs(rawOutputs.concat(outputs))),
+      },
+    });
+
+    const transaction = await this.provider.getTransaction(from.network, txid);
+    if (transaction.type !== 'InvocationTransaction') {
+      throw new InvocationCallError(`Expected InvocationTransaction, received ${transaction.type}`);
+    }
+
+    return {
+      transaction,
+      confirmed: async () => {
+        const [receipt, data] = await Promise.all([
+          this.provider.getTransactionReceipt(from.network, txid),
+          this.provider.getInvocationData(from.network, txid),
+        ]);
+
+        return onConfirm({ transaction, receipt, data });
+      },
+    };
+  }
+
+  private convertDapiAttributes(attributes: readonly Attribute[]): readonly TxHashAttribute[] {
+    return attributes.map((attribute) => {
+      const argBase = paramToArgDataType(attribute.data);
+
+      return {
+        ...argBase,
+        txAttrUsage: attribute.usage,
+      };
+    });
+  }
+
+  private async init(): Promise<void> {
+    await new Promise<void>((resolve) => {
```

Comment:
This needs a timeout, otherwise the application might just hang forever when they call a method that depends on this.